scheduler/src/cook/config.clj

```diff
@@ -429,8 +438,8 @@
                                   :default-pool "no-pool"}
                                  pool-selection)})))
      :kubernetes (fnk [[:config {kubernetes {}}]]
-                   (merge kubernetes
-                          {:default-workdir "/mnt/sandbox"}))}))
+                   (merge {:default-workdir "/mnt/sandbox"}
+                          (update kubernetes :pod-ip->hostname-fn resolve-optional-function identity)))}))
```

Comment:
This seems very weird. Like iits wrapping an inner configuration. I think I know whats going on here, you should deconstruct the kubernetes using fnk, instead of this update.

Reply:
will refactor later